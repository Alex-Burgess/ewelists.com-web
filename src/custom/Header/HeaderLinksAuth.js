import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// react components for routing our app without refresh
import { Link } from "react-router-dom";

// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
// @material-ui/icons
import AccountCircle from "@material-ui/icons/AccountCircle";
import ExitToApp from "@material-ui/icons/ExitToApp";
import Settings from "@material-ui/icons/Settings";
// core components
import CustomDropdown from "components/CustomDropdown/CustomDropdown.js";

import styles from "assets/jss/material-kit-pro-react/components/headerLinksStyle.js";
const useStyles = makeStyles(styles);

export default function HeaderLinks(props) {
  const classes = useStyles();
  const { dropdownHoverColor, user } = props;

  return (
    <List className={classes.list + " " + classes.mlAuto}>
      <ListItem className={classes.listItem}>
        <Link to="/" className={classes.navLink}>
          My Lists
        </Link>
      </ListItem>
      <ListItem className={classes.listItem}>
        <Link to="/list-ideas" className={classes.navLink}>
          Gift List Ideas
        </Link>
      </ListItem>
      <ListItem className={classes.listItem}>
        <CustomDropdown
          noLiPadding
          navDropdown
          hoverColor={dropdownHoverColor}
          buttonProps={{
            className: classes.navLink,
            color: "transparent"
          }}
          buttonIcon={Settings}
          dropdownList={[
            <Link to="/" className={classes.dropdownLink}>
              <AccountCircle className={classes.dropdownIcons} /> {user.name}
            </Link>,
            <Link to="/logout" className={classes.dropdownLink}>
              <ExitToApp className={classes.dropdownIcons} /> Sign Out
            </Link>
          ]}
        />
      </ListItem>
    </List>
  );
}

HeaderLinks.defaultProps = {
  hoverColor: "primary"
};

HeaderLinks.propTypes = {
  dropdownHoverColor: PropTypes.oneOf([
    "dark",
    "primary",
    "info",
    "success",
    "warning",
    "danger",
    "rose"
  ])
};
