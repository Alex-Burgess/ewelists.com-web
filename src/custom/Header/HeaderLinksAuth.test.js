import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import HeaderLinksAuth from './HeaderLinksAuth';

test('Renders HeaderLinksAuth component, when user is not authenticated.', () => {
  const tree = renderer
    .create(
      <Router>
        <HeaderLinksAuth dropdownHoverColor="info" />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
