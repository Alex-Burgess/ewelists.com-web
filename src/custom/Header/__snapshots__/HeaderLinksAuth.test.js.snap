// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders HeaderLinksAuth component, when user is not authenticated. 1`] = `
<ul
  className="MuiList-root makeStyles-list-1 makeStyles-mlAuto-17 MuiList-padding"
>
  <li
    className="MuiListItem-root makeStyles-listItem-2 MuiListItem-gutters"
    disabled={false}
  >
    <a
      className="makeStyles-navLink-4"
      href="/"
      onClick={[Function]}
    >
      My Lists
    </a>
  </li>
  <li
    className="MuiListItem-root makeStyles-listItem-2 MuiListItem-gutters"
    disabled={false}
  >
    <a
      className="makeStyles-navLink-4"
      href="/list-ideas"
      onClick={[Function]}
    >
      Gift List Ideas
    </a>
  </li>
  <li
    className="MuiListItem-root makeStyles-listItem-2 MuiListItem-gutters"
    disabled={false}
  >
    <div
      className="makeStyles-manager-35"
    >
      <div
        className="makeStyles-target-37"
      >
        <button
          aria-haspopup="true"
          aria-label="Notifications"
          aria-owns={null}
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-58 makeStyles-transparent-81 makeStyles-navLink-4"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root makeStyles-buttonIcon-51"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h20v20H0V0z"
                fill="none"
                transform="scale(1.2, 1.2)"
              />
              <path
                d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
                transform="scale(1.2, 1.2)"
              />
            </svg>
            <b
              className="makeStyles-caret-52"
            />
          </span>
        </button>
      </div>
    </div>
  </li>
</ul>
`;
